src/redux/actions/cellSets/loadCellSets.js

```diff
@@ -5,15 +5,19 @@ import {
 
 const loadCellSets = (experimentId) => async (dispatch, getState) => {
   const {
-    loading, error,
+    loading, error, updatingClustering,
   } = getState().cellSets;
 
-  if (!loading && !error) {
+  if ((!loading && !error) || updatingClustering) {
     return null;
   }
   // There is only two way this action dispatcher does anything. Either
   // it is called after an error condition, or in a loading state. In the former
   // state, dispatching the loading action is unnecessary.
+
+  // Apart from this, if the clustering is being updated,
+  // this action dispatcher ends too since we will get a more up to date when the task is done
+
```

Comment:
These comments are not great:
The first one is hard to read and super confusing - I had to read it like 3 times in order to understand what it talks about. And when I understood I was like "cool, so what?". For the bit that says `In the former state, dispatching the loading action is unnecessary.` if it is unnecessary, then why are we dispatching it anyway? The code here is quite short and simple, I don't think we need a comment here at all. Moreover, I think that this first comment only adds more confusion as to what is happening - makes it look like something very complex is going on, when it actually isn't. If I was you, I would just delete it.

For the second comment - I would move the comment to be before the if statement on line 11, because this is technically when the action will end by returning null.

Reply:
Yes, I found the previous comments confusing too, I'll delete it.

About my own added comment, should I delete it too since it is nothing particularly complex?

Reply:
yup, you can delete yours too